src/util/rule.ts

```diff
@@ -11,6 +11,17 @@ import type { Node as TSNode, Type } from "typescript";
 // eslint-disable-next-line @typescript-eslint/naming-convention, no-underscore-dangle -- This is a special var.
 const __VERSION__ = "0.0.0-development";
 
+/**
+ * Any custom rule meta properties.
+ */
+export type NamedCreateRuleMetaWithCategory<T extends string> =
+  ESLintUtils.NamedCreateRuleMeta<T> & {
+    docs: {
+      /** Used for splitting the README rules list into sub-lists. */
+      category: string;
```

Comment:
Had to add this property to the rule meta object since we use it for splitting the rule lists. This is not an official ESLint property anymore but still commonly used by plugins.